lib/wibox/container/background.lua

```diff
@@ -268,20 +268,16 @@ function background:get_shape_border_color()
     return self._private.shape_border_color
 end
 
---- When a `shape` is set, make sure nothing is drawn outside of it.
---@DOC_wibox_container_background_clip_EXAMPLE@
--- @property shape_clip
--- @tparam boolean value If the shape clip is enable
-
 function background:set_shape_clip(value)
-    if self._private.shape_clip == value then return end
-
-    self._private.shape_clip = value
-    self:emit_signal("widget::redraw_needed")
+    if value then return end
+    require("gears.debug").print_warning("shape_clip property of background container was removed."
```

Comment:
This is removing, not deprecating as the commit says, `shape_clip`?

Reply:
Two commits before that (the commit that does "the big rewrite") does the actual removing. This commit just turns "silently ignore this option" into "un-silently ignore this option".
So, this is only removing the internal implementation of the setter, but its "outside observable" behaviour is basically the same (the only observable difference is that after setting the option to `false`, the getter will still return `true`; no visual differences).

Reply:
Ah! Understood. Just used to seeing a deprecation warning rather than a removed one. :)